packages/google_maps_flutter/google_maps_flutter/example/integration_test/src/maps_inspector.dart

```diff
@@ -612,6 +627,186 @@ void runTests() {
+        onCameraIdle: () {
+          cameraIdleFired = true;
+          cameraFiredStream.add(null);
+        },
+        onMapCreated: (GoogleMapController controller) {
+          controllerCompleter.complete(controller);
+        },
+      ),
+    ));
+
+    final GoogleMapController controller = await controllerCompleter.future;
+
+    await tester.pumpAndSettle();
+    // TODO(cyanglaz): Remove this after we added `mapRendered` callback, and `mapControllerCompleter.complete(controller)` above should happen
+    // in `mapRendered`.
+    // https://github.com/flutter/flutter/issues/54758
+    await Future<void>.delayed(const Duration(seconds: 1));
+
+    // Drain any event that might have been added from the initial setup or
+    // resetting the camera position.
+    if (cameraIdleFired) {
+      await cameraFiredQueue.next;
+      cameraIdleFired = false;
+    }
+
+    final CameraUpdate cameraUpdate =
+        _getCameraUpdateForType(_cameraUpdateTypeVariants.currentValue!);
+    await controller.animateCamera(cameraUpdate);
+
+    // Check that position is not updated immediately to the target.
+    final CameraPosition beforeFinishedPosition =
+        await inspector.getCameraPosition(mapId: controller.mapId);
+    await _checkCameraUpdateByType(
+        _cameraUpdateTypeVariants.currentValue!,
+        beforeFinishedPosition,
+        null,
+        controller,
+        (Matcher matcher) => isNot(matcher));
+
+    // Check that position is animated after the animation is done.
+    expect(cameraIdleFired, isFalse);
+    await cameraFiredQueue.next;
+    expect(cameraIdleFired, isTrue);
+    final CameraPosition afterFinishedPosition =
+        await inspector.getCameraPosition(mapId: controller.mapId);
+    await _checkCameraUpdateByType(
+        _cameraUpdateTypeVariants.currentValue!,
+        afterFinishedPosition,
+        beforeFinishedPosition,
+        controller,
+        (Matcher matcher) => matcher);
+
+    await tester.pumpAndSettle();
+  }, variant: _cameraUpdateTypeVariants);
+
+  testWidgets(
+    'testAnimateCameraWithConfiguration',
+    (WidgetTester tester) async {
+      final Key key = GlobalKey();
+      final Completer<GoogleMapController> controllerCompleter =
+          Completer<GoogleMapController>();
+      final GoogleMapsInspectorPlatform inspector =
+          GoogleMapsInspectorPlatform.instance!;
+
+      final StreamController<void> cameraFiredStream = StreamController<void>();
+      final StreamQueue<void> cameraFiredQueue =
+          StreamQueue<void>(cameraFiredStream.stream);
+
+      // We set the duration to 100ms and check that the animation is completed
+      // faster than the normal animation (~300ms on Android) to make sure the
+      // configuration is really used.
+      const int cameraAnimationDurationMS = 100;
+
+      // The threshold is set to 200ms because the normal camera animation should
+      // take more than 200ms on android.
```

Comment:
This is very likely to become a flaky test as written, since it assumes things about the timing of things beyond the animation itself (multiple async steps with cross-thread hops, potentially delays within the Google Maps SDK itself), and the tolerance here is relatively tight for environments like an emulator. It's also not ideal (per the previous comment) to have a test that would become subject to false negatives if a value we don't control in a different SDK changed.

What about having the test run once with an extremely short time, like 10ms, and once with a much longer time like a second, and make sure the difference between them is more than, say, half the larger value. That gives a fair amount of tolerance, but should still easily catch the value being ignored.

Reply:
I used suggested 10ms and 1000ms durations and tests seems to pass.
But the duration increase caused CI tests to timeout
@stuartmorgan, how to proceed with this?

Reply:
Increasing a few integration test durations by a second shouldn't have caused timeouts; are you sure there wasn't some other change that's causing timeouts now?

Reply:
@stuartmorgan 
Is there a way to access more detailed test output than this: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8729373696952662689/+/u/Run_package_tests/Firebase_Test_Lab/stdout
(I don't have access to the firebase console: https://console.firebase.google.com/project/flutter-infra-staging/testlab/histories/bh.8eb89ae93eb3eb36/matrices/6375674335592449274)

Tests run fine locally without timeouts.

Running tests for all CameraUpdateType variants (9 types) adds at least 12 seconds per package due to the testing with 200ms and 1000ms animation delays + map initialization. 

Should we consider testing just one type here since implementations between CameraUpdate types are identical on iOS and Android?
On web implementation testing each variation is more important because there is separate method for each camera update type.

Reply:
Unfortunately the console isn't showing anything interesting that I can see. The way integration tests run in FTL we don't get incremental test output; so it's mostly device-level logs. For the rare cases like this where an issue happens in FTL but not locally, sometimes we have to do guess-and-check debugging unfortunately, by pushing alternate versions (I would suggest a new PR so you don't pollute this one with unwanted changes) with changes. E.g., try just a minimal new test, see if it works, then add more incrementally. It's not ideal, but the FTL-style run just doesn't always give us the info we need.

You could also try adding some native logging so that we can look for that in the device logs, to get a sense of what is happening on the native side. E.g., maybe log before each animated move request, with some details about the args to try to identify it, and also before sending each camera idle notification to Dart.

> Running tests for all CameraUpdateType variants (9 types) adds at least 12 seconds per package due to the testing with 200ms and 1000ms animation delays + map initialization.
> 
> Should we consider testing just one type here since implementations between CameraUpdate types are identical on iOS and Android?

We could, but for context the end-to-end integration test run for `google_maps_flutter_android` right now in CI is about 6 minutes, so 12 seconds of additional tests aren't really that big a deal.

Reply:
Tested locally 
- with one CameraUpdate variant, all tests runned in average 48 seconds. (tested 3 times)
- with all 9 CameraUpdate variants, all tests runned in average 1 minute 22 seconds (tested 3 times)

I bet there is 1 minute timeout set for the tests.

Should we remove variants and test only with one CameraUpdate type?
@stuartmorgan ?

Reply:
I was curious, so I just pulled this branch and tried it locally, and see it hang in my simulator. I'll debug a bit and see what I find.

Reply:
What I see locally is that `await cameraFiredQueue.next;` never completes.

Are you testing with a maps key? We don't currently test with one in CI because in practice everything we were testing worked without it, but maybe this is something that doesn't work without a key?

Reply:
@stuartmorgan 
`animateCamera` does not move the camera at all on Android if Google Maps API key is not given. 

Should we skip these tests on android for now?

Reply:
In theory yes (https://github.com/flutter/flutter/issues/131071), but it will require some coordination with infra-level changes so it's not something I expect to happen in the very short term. We could try to prioritize it in Q1 of next year, potentially.

For now it's fine to skip this test on Android with a reference to that issue in a comment.

Reply:
Skipping tests on Android
Tests works fine with api key, when enabled.